cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/common/DBRecord.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.common;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.collect.Lists;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.lib.db.DBWritable;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Writable class for DB Source/Sink
+ *
+ * {@see DBWritable}, {@see DBInputFormat}, {@see DBOutputFormat}
+ */
+public class DBRecord implements Writable, DBWritable {
+
+  private StructuredRecord record;
+
+  public void readFields(DataInput in) throws IOException {
+  }
+
+  /**
+   * @return the {@link StructuredRecord} contained in this object
+   */
+  public StructuredRecord getRecord() {
+    return record;
+  }
+
+  /**
+   * Builds the {@link #record} using the specified {@link ResultSet}
+   *
+   * @param resultSet the {@link ResultSet} to build the {@link StructuredRecord} from
+   */
+  public void readFields(ResultSet resultSet) throws SQLException {
+    List<Schema.Field> fields = Lists.newArrayList();
+    ResultSetMetaData metadata = resultSet.getMetaData();
+    // ResultSetMetadata columns are numbered starting with 1
+    for (int i = 1; i <= metadata.getColumnCount(); i++) {
+      String columnName = metadata.getColumnName(i);
+      String columnType = metadata.getColumnClassName(i);
+      Schema.Field field = Schema.Field.of(columnName, Schema.of(getType(columnType)));
```

Comment:
Isn't that you have the check if the column type is nullable?

http://docs.oracle.com/javase/6/docs/api/java/sql/ResultSetMetaData.html#field_detail